anyway/infographics_utils.py

```diff
@@ -949,25 +949,105 @@ class MotorcycleAccidentsVsAllAccidentsWidget(Widget):
     def __init__(self, request_params: RequestParams):
         super().__init__(request_params, WidgetId.motorcycle_accidents_vs_all_accidents)
         self.rank = 20
-        self.text = {"title": "תאונות אופנועים קשות וקטלניות בכביש 20 בהשוואה לכל הארץ"}
+        self.road_number = request_params.location_info["road1"]
+        self.text = {"title": f"תאונות אופנועים קשות וקטלניות בכביש {int(self.road_number)} בהשוואה לכל הארץ"}
 
     # noinspection PyMethodMayBeStatic
     def is_in_cache(self) -> bool:
         return False
 
     def generate_items(self) -> None:
         self.items = (
-            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents_mock_data()
+            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents(
+                self.request_params.start_time, self.request_params.end_time, self.road_number)
         )
 
+    '''with ROAD_NUMBER as (
+        select 20 as val
+    )
+    SELECT 
+    LOCATION, vehicle, num_of_accidents/sum(num_of_accidents) 
+        over(PARTITION BY LOCATION):: decimal AS percentage 
+    FROM 
+    (SELECT CASE WHEN road1=(select * from ROAD_NUMBER) or road2 = (select * from ROAD_NUMBER) 
+                    THEN CONCAT('כביש ',(select * from ROAD_NUMBER)::text)  
+                 ELSE 'כל הארץ' END AS LOCATION, 
+    -- grouping of vehicles 
+    CASE WHEN involve_vehicle_type IN (8, 10, 19, 9) THEN 'אופנוע' ELSE 'אחר' END AS vehicle,
+    COUNT(DISTINCT provider_and_id) num_of_accidents 
+    FROM involved_markers_hebrew 
+    WHERE road_type in (3,4) -- only inter-city 
+    AND accident_severity in (1,2) GROUP BY 1,2 ORDER BY 3 DESC) as ACCIDENTS_COUNT'''
```

Comment:
I do not understand the syntax :(, is this a docstring?, If yes, of the above or below

Reply:
No, A docstring is a string literal that occurs as the first statement in a module, function, class, or method definition. 
For example:
def foo():
    """Return the number 1."""
    return 1

This is the SQL query that should describe the function functionality - I agree with you that it's not very readable,
If you have any question about the query I can try to explain(I didn't write it so I am not fully sure about some of code in this query)

Reply:
The reason I left it there is that when I was trying to figure out how to implement it I had a hard time learning from existing code since I couldn't easily tell what the sqlalchemy code was achieving in sql vocabulary.
So I thought it may be useful to leave inside.
I can delete it if you don't see any values in it.

Reply:
Yes it will be better without it - we have the Pull request comments and the issue for this info

Reply:
ok